server/apps/publish/published_item.py

```diff
@@ -472,17 +492,52 @@ def __get_user_name(self, user_id):
         Retrieves display_name of the user identified by user_id
         """
 
-        if user_id == '':
+        if not user_id:
             return ''
 
         user = get_resource_service('users').find_one(req=None, _id=user_id)
         return get_display_name(user)
 
-    def __is_orphan(self, doc):
+    def can_remove_from_production(self, doc):
         """
-        Checks if the doc in published collection is orphan in archive collection or not.
+        Returns true if the doc in published collection can be removed from production, otherwise returns false.
+        1. Returns false if item is published more than once
+        2. Returns false if item is referenced by a package
+        3. Returns false if the item is package and all items in the package are not found in archived collection.
+
         :param doc: article in published collection
-        :return: True if orphan in archive collection, False otherwise.
+        :return: True if item can be removed from production, False otherwise.
         """
 
-        return PackageService().get_packages(doc['item_id']).count() == 0
+        items = self.get_other_published_items(doc['item_id'])
+        is_removable = (items.count() == 0)
+
+        if is_removable:
+            is_removable = (PackageService().get_packages(doc['item_id']).count() == 0)
+
+            if is_removable and doc[ITEM_TYPE] == CONTENT_TYPE.COMPOSITE:
+                return self._can_remove_package_from_production(doc)
+
+        return is_removable
+
+    def _can_remove_package_from_production(self, package):
+        """
+        Recursively checks if the package can be removed from production.
+
+        :param package:
+        :return: True if item can be removed from production, False otherwise.
+        """
+
+        item_refs = PackageService().get_residrefs(package)
+        archived_items = list(get_resource_service('archived').find_by_item_ids(item_refs))
+        is_removable = (len(item_refs) == len(archived_items))
+
+        if is_removable:
+            packages_in_archived_items = [p for p in archived_items if p[ITEM_TYPE] == CONTENT_TYPE.COMPOSITE]
+
+            for package in packages_in_archived_items:
+                is_removable = self._can_remove_package_from_production(package)
+                if is_removable is False:
```

Comment:
Probably more Pythonic to say `if not is_removable`, `is` is more for comparing object references (although it works here, too).